spartan_parallel/src/sparse_mlpoly.rs

```diff
@@ -436,25 +369,21 @@ impl SparseMatPolynomial {
         let row = self.M[i].row;
         let col = self.M[i].col;
         let val = &self.M[i].val;
-        eval_table_rx[row] * eval_table_ry[col] * val
+        eval_table_rx[row] * eval_table_ry[col] * *val
```

Comment:
If we are implementing our own field element arithmetic anyway, we can add instances for arithmetic with references, too.  See how we do that in Ceno.

(We don't need to do that in this PR.)